flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SchedulerUtilsTest.java

```diff
@@ -40,16 +69,55 @@ public void testSettingMaxNumberOfCheckpointsToRetain() throws Exception {
         jobManagerConfig.setInteger(
                 CheckpointingOptions.MAX_RETAINED_CHECKPOINTS, maxNumberOfCheckpointsToRetain);
 
-        final CompletedCheckpointStore completedCheckpointStore =
-                SchedulerUtils.createCompletedCheckpointStore(
-                        jobManagerConfig,
-                        getClass().getClassLoader(),
-                        new StandaloneCheckpointRecoveryFactory(),
-                        log,
-                        new JobID());
-
-        assertEquals(
-                maxNumberOfCheckpointsToRetain,
-                completedCheckpointStore.getMaxNumberOfRetainedCheckpoints());
+        final CompletedCheckpointStore actualCompletedCheckpointStore =
+                SchedulerUtils.createCompletedCheckpointStoreIfCheckpointingIsEnabled(
+                        jobGraph, jobManagerConfig, classLoader, checkpointRecoveryFactory, log);
+
+        assertTrue(expectedCompletedCheckpointStore.recovered);
+        assertEquals(expectedCompletedCheckpointStore, actualCompletedCheckpointStore);
+    }
+
+    private class RecoveryRecordingCompletedCheckpointStore implements CompletedCheckpointStore {
```

Comment:
This can be simplified by just extending StandalonCompletedCheckpointStore, then you just need to override a single method. Also this class can be static.